generate_schedule.py

```diff
@@ -78,7 +74,12 @@ def clamp_to_gc(power):
     }
 
     # get battery info: how much can be discharged in beginning, how much if fully charged?
-    batteries = s.world_state.batteries.values()
+    batteries = {}
+    for battery, values in s.world_state.batteries.items():
+        if s.world_state.batteries[battery].parent == list(
+                scenario.constants.grid_connectors)[gc_index]:
+            batteries.update({battery: values})
+    batteries = batteries.values()
```

Comment:
This is needlessly complicated. Suggestion: `batteries = [b for b in s.world_state.batteries.values() if b.parent == gcID]`